ResearchKit/Common/ORKFormItemCell.m

```diff
@@ -469,26 +488,59 @@ - (void)answerDidChange {
     }
 }
 
+- (BOOL)isAnswerValid {
+    NSString *text = self.textField.text;
+    BOOL isValid = YES;
+    if ([text length]) {
+        isValid = [[self.formItem impliedAnswerFormat] isAnswerValidWithString:text];
+    }
+    return isValid;
+}
+
 #pragma mark UITextFieldDelegate
 
+- (BOOL)textFieldShouldEndEditing:(UITextField *)textField {
+    NSString *text = textField.text;
+    BOOL isValid = [self isAnswerValid];
+    if (! isValid) {
+        [self showValidityAlertWithMessage:text];
+    }
+    return YES;
+}
+
 - (void)textFieldDidEndEditing:(UITextField *)textField {
     [super textFieldDidEndEditing:textField];
     [self inputValueDidChange];
 }
 
+- (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    BOOL isValid = [self isAnswerValid];
+    
+    if (! isValid) {
+        [self showValidityAlertWithMessage:[[self.formItem impliedAnswerFormat] localizedInvalidValueStringWithAnswerString:textField.text]];
```

Comment:
If `localizedInvalidValueStringWithAnswerString` returns nil for valid string, then `[self isAnswerValid]` is not necessary.